### PR TITLE
[Merged by Bors] - feat: add some Associated/Prime lemmas

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
From flt-regular.
    
Co-authored-by: Andrew Yang <the.erd.one@gmail.com>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/BigOperators/Associated.lean [hunk 1]
```diff
@@ -57,6 +57,19 @@ theorem Prod.associated_iff {M N : Type*} [Monoid M] [Monoid N] {x z : M × N} :
   fun ⟨⟨u₁, h₁⟩, ⟨u₂, h₂⟩⟩ =>
     ⟨MulEquiv.prodUnits.invFun (u₁, u₂), Prod.eq_iff_fst_eq_snd_eq.2 ⟨h₁, h₂⟩⟩⟩
 
+theorem Associated.prod {M : Type*} [CommMonoid M] {ι : Type*} (s : Finset ι) (f : ι → M)
+    (g : ι → M) (h : ∀ i, i ∈ s → Associated (f i) (g i)) : (∏ i in s, f i) ~ᵤ (∏ i in s, g i) := by
```
Comment (ocfnash) [inline Mathlib/Algebra/BigOperators/Associated.lean:61]:
```suggestion
    (g : ι → M) (h : ∀ i, i ∈ s → (f i) ~ᵤ (g i)) : (∏ i in s, f i) ~ᵤ (∏ i in s, g i) := by
```

File: Mathlib/Algebra/BigOperators/Associated.lean [hunk 2]
```diff
@@ -57,6 +57,19 @@ theorem Prod.associated_iff {M N : Type*} [Monoid M] [Monoid N] {x z : M × N} :
...
+  induction s using Finset.induction with
+  | empty =>
+    simp only [Finset.prod_empty]
+    rfl
+  | @insert j s hjs IH =>
+    classical
+    convert_to Associated (∏ i in insert j s, f i) (∏ i in insert j s, g i)
```
Comment (ocfnash) [inline Mathlib/Algebra/BigOperators/Associated.lean:68]:
```suggestion
    convert_to (∏ i in insert j s, f i) ~ᵤ (∏ i in insert j s, g i)
```

**Review by ocfnash — APPROVED** [review]
Thanks!

bors d+

**Comment by Ruben-VandeVelde** [comment]
bors merge

**Pull request closed without merging**